share/getters/options.go

```diff
@@ -0,0 +1,20 @@
+package getters
+
+import (
+	"github.com/celestiaorg/celestia-node/pruner"
+	"github.com/celestiaorg/celestia-node/share/p2p/peers"
+)
+
+type Option func(*ShrexGetter)
+
+func WithArchivalPeerManager(manager *peers.Manager) func(*ShrexGetter) {
+	return func(s *ShrexGetter) {
+		s.archivalPeerManager = manager
+	}
+}
+
+func WithAvailabilityWindow(window pruner.AvailabilityWindow) func(*ShrexGetter) {
+	return func(s *ShrexGetter) {
+		s.availabilityWindow = window
+	}
+}
```

Comment:
Looking at this codes makes me wonder what is the case when archivalPeerManager is nil? It seems it should always be provided, to allow shrex.Getter to access historcal data. Lets make it archivalPeerManager and availabilityWindow part of constructor and not an option

Reply:
At the time, I was unsure about whether every node needs to have an archival service, but now we said that they need to.